meta-ov/recipes-apps/ovmenu-ng-skripts/files/transfer-xcsoar.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Transfer script for Up/Downloadng data to usbstick
+
+USB_PATH="/usb/usbstick/openvario/"
+XCSOAR_PATH="/home/root/.xcsoar"
+
+case "$(basename $0)" in
+  'download-all.sh')
+    SRC_PATH="$XCSOAR_PATH"
+    DEST_PATH="$USB_PATH/download/xcsoar"
+    ;;
+  'upload-xcsoar.sh')
+    SRC_PATH="$USB_PATH/upload/xcsoar"
+    DEST_PATH="$XCSOAR_PATH"
+    ;;
+  'upload-all.sh')
+    SRC_PATH="$USB_PATH/upload"
+    DEST_PATH="$XCSOAR_PATH"
+    ;;
+  *)
+    echo 'call as download-all.sh, upload-xcsoar.sh or upload-all.sh'
+    exit 1
+esac
+
+if [ ! -d "$SRC_PATH" ] || [ ! -d "$DEST_PATH" ]; then
+  echo "Source $SRC_PATH or destination path $DEST_PATH does not exist"
+  exit 1
+fi
+
+if [ -z "$(find "$SRC_PATH" -type f | head -n 1 2>/dev/null)" ]; then
+  echo 'No files found !!!'
+else
+  # echo and copy each file / directory for tui
+  cd "$SRC_PATH" || exit 1
+  find ./ -type d -exec echo {} \; -exec mkdir -p "$DEST_PATH/{}" \;
+  if find ./ -type f -exec echo {} \; -exec cp "{}" "$DEST_PATH/{}" \; ; then
```

Comment:
Instead of checks like that, I think it would be simpler to just use `set -e`

Reply:
Yes, but then it would be hard to display an end-user readable error message.

Reply:
I thought the error message printed by `cp` was already good enough.
But I just found out that `find` does not propagate `-exec` failures to the caller. It just keeps on executing more commands even if one has failed, and exits with status 0 (success). Did you know that? (I didn't because I never use `find -exec`)

Reply:
It aborts on first error, yet the return code is 0. What a dangerous design. 
```shell
touch 1 2 3
find -exec cat "{}" \;
cat: ./: Is a directory
echo $?
0
```
So my if check statement doesn't catch what i thought it would.

There is however the -exec {} +  form, that one would return a non-zero value from an execd process. 

So in the end this too much complexity. I will redesign.

Reply:
So my new approach uses rsync. I know 499kb + , but:
The transfers to the stick now take mere seconds (even with checksumming) for a 400MB takes a mere 20seconds (with a previous copy on the stick) as compared for the about 20minutes with the previous solution.
I took care to use -c for checksum comparision  only, ignoring filesize and timestamps.